src/bt.py

```diff
@@ -0,0 +1,106 @@
+import struct
+import binascii
+import bluetooth
+from micropython import const
+import ble_advertising as blea
+
+
+_IRQ_SCAN_RESULT = const(5)
+_IRQ_SCAN_DONE = const(6)
+
+_GAP_SCAN_INTERVAL_US = const(128_000)
+_GAP_SCAN_WINDOW_US = const(11_250)
+_GAP_ADV_INTERVAL_US = const(100_000)
+
+class LampBluetooth:
+    MAGIC_NUM = const(42069)
+    COLORS_FIELD_FORMAT = "<H3B3B"
+    @classmethod
+    def _hex_to_rgb(cls, color: str):
+        hex_color = color.lstrip('#')
+        return tuple(int(hex_color[i:i+2], 16) for i in (0, 2, 4))
+
+    @classmethod
+    def _rgb_to_hex(cls, rgb_color):
```

Comment:
Maybe when tying this in, we should move these to a util file so that we can use them in led_strip as well, and nuke the hext2rgb that's present there